File: back-end/src/main/java/com/joaoalencar/projetoimpacta/domain/model/product/Product.java
package com.joaoalencar.projetoimpacta.domain.model.product;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Entity
@Table(name = "products")
@Getter
@Setter
public class Product {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "price")
    private float price;

    @Column(name = "stock")
    private int stock;

    @Column(name = "supplier_id")
    private Integer supplierId;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "product")
    private List<ProductImage> images;
}

File: api/Domain/Entities/Product.cs
namespace api.Domain.Entities
{
    public class Product
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Stock { get; set; }
        public long SupplierId { get; set; }

        public Product()
        {
        }

        public Product(string name, string description, decimal price, int stock, long supplierId)
        {
            Name = name;
            Description = description;
            Price = price;
            Stock = stock;
            SupplierId = supplierId;
        }

        public Supplier Supplier { get; set; } = default!;
        public ICollection<ProductImage> Images  { get; set; } = default!;
    }
}